test/unit/test_idstorage.c

```diff
@@ -0,0 +1,71 @@
+#include <r_util.h>
+#include "minunit.h"
+
+bool test_r_id_storage_add(void) {
+	char *str = "lol";
+	RIDStorage *ids = r_id_storage_new (5, 23);
+	ut32 id;
+	r_id_storage_add (ids, str, &id);
+	void *ptr = r_id_storage_get (ids, id);
+	r_id_storage_free (ids);
+	mu_assert_ptreq (ptr, str, "id_storage_add");
+	mu_end;
+}
+
+bool test_r_id_storage_set(void) {
+	char *str = "lol";
+	RIDStorage *ids = r_id_storage_new (5, 23);
+	r_id_storage_set (ids, str, 1);
+	void *ptr = r_id_storage_get (ids, 1);
+	r_id_storage_free (ids);
+	mu_assert_ptreq (ptr, str, "id_storage_set");
+	mu_end;
+}
+
+bool test_r_id_storage_delete(void) {
+	RIDStorage *ids = r_id_storage_new (5, 23);
+	ut32 id;
+	r_id_storage_add (ids, "lol", &id);
+	r_id_storage_delete (ids, id);
+	void *ptr = r_id_storage_get (ids, id);
+	r_id_storage_free (ids);
+	mu_assert_ptreq (ptr, NULL, "id_storage_delete");
+	mu_end;
+}
+
+bool test_r_id_storage_take0(void) {
+	char *str = "lol";
+	RIDStorage *ids = r_id_storage_new (5, 23);
+	ut32 id;
+	r_id_storage_add (ids, str, &id);
+	void *ptr = r_id_storage_take (ids, id);
```

Comment:
Difference between get and take?

Reply:
get gets you the element that is adressed by the id.
take does the same but also deletes the element from the idstorage

Reply:
Ok thats what i was expecting too but is the only place in r2land where this verb is used